core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -140,11 +146,14 @@ protected FileInStream(URIStatus status, InStreamOptions options, FileSystemCont
     mShouldCache = mAlluxioStorageType.isStore();
     mShouldCachePartiallyReadBlock = options.isCachePartiallyReadBlock();
     mClosed = false;
-    mLocationPolicy = options.getLocationPolicy();
+    mCacheLocationPolicy = options.getLocationPolicy();
```

Comment:
yeah, in order to change this, i need to deprecate getLocationPolicy() which is a public API. if you are fine with updating or deprecating that. i can change it.  The complexity of deprecating this is to make our REST API backward compatible.